File: llm_generated_code/ChatGPT/Python/Level1/Q27.py
def is_equilibrium(forces):
    sum_x = sum(force[0] for force in forces)
    sum_y = sum(force[1] for force in forces)
    sum_z = sum(force[2] for force in forces)
    return sum_x == 0 and sum_y == 0 and sum_z == 0

forces = []
n = int(input())
for _ in range(n):
    forces.append(tuple(map(int, input().split())))
result = "YES" if is_equilibrium(forces) else "NO"
print(result)

File: llm_generated_code/ChatGPT/CPP/Level1/Q27.cpp
#include <iostream>
#include <vector>

using namespace std;

struct Force {
    int x, y, z;
};

bool isEquilibrium(const vector<Force>& forces) {
    int sumX = 0, sumY = 0, sumZ = 0;
    for (const auto& force : forces) {
        sumX += force.x;
        sumY += force.y;
        sumZ += force.z;
    }
    return (sumX == 0 && sumY == 0 && sumZ == 0);
}

int main() {
    int n;
    cin >> n;
    vector<Force> forces(n);
    for (int i = 0; i < n; ++i) {
        cin >> forces[i].x >> forces[i].y >> forces[i].z;
    }
    cout << (isEquilibrium(forces) ? "YES" : "NO") << endl;
    return 0;
}
